src/System.Text.Json/tests/Utf8JsonReaderTests.cs

```diff
@@ -1770,66 +1770,89 @@ public static void ConsumeLotsOfComments(string valueString, bool insideArray, s
             Assert.Equal(dataUtf8.Length, json.BytesConsumed);
         }
 
+        struct SingleCommentVerifyInfo
+        {
+            public bool startObjectFound;
+            public bool endObjectFound;
+            public bool commentFound;
+        }
+
+        private static void VerifyReadLoop(ref Utf8JsonReader json, string expected, ref SingleCommentVerifyInfo verifyInfo)
+        {
+            while (json.Read())
+            {
+                switch (json.TokenType)
+                {
+                    case JsonTokenType.StartObject:
+                        verifyInfo.startObjectFound = true;
+                        break;
+                    case JsonTokenType.EndObject:
+                        verifyInfo.endObjectFound = true;
+                        break;
+                    case JsonTokenType.Comment:
+                        expected?.Equals(Encoding.UTF8.GetString(json.ValueSpan));
```

Comment:
I think we just need to assert that we found the matching comment. Verifying that `commentFound == true` is redundant.

For skipping, we could instead pass a bool to this helper and Assert.False() if we hit the Comment TokenType.

In fact, we could use `expected == null` as the bool:
```C#
if (expected == null) Assert.False()
else Assert.Equals(expected, Encoding.UTF8.GetString(json.ValueSpan))
```

Reply:
Oh ok. I remember. I added `commentFound == true` because if in a `Consume...` test case, I pass flag `Skip` (I had done that typo), then still the test case succeeds, because then `case JsonTokenType.Comment` is never encountered, although `expected` is not null. It was just making the test case robust. And then I extended all that to start and end object as well to confirm json integrity. Thoughts?

Reply:
I don't have a strong opinion, but trying to test the correctness of a test itself (to avoid potential typo/bugs) as part of the test seems unnecessary. We want to test correctness of the API itself (the particular functionality). If, you want to add extra validation, its fine, but creating a struct/etc. complicates the test code for little value, imo.